core/meta/src/TClass.cxx

```diff
@@ -5464,20 +5464,15 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
+   // Return early if class info cannot be loaded or is already loaded.
+   if (!fCanLoadClassInfo) return;
+
    R__LOCKGUARD(gInterpreterMutex);
 
    // If another thread executed LoadClassInfo at about the same time
    // as this thread return early since the work was done.
    if (!fCanLoadClassInfo) return;
 
-   // If class info already loaded then do nothing.  This can happen if the
-   // class was registered by a dictionary, but the info came from reading
-   // the pch.
-   // Note: This check avoids using AutoParse for classes in the pch!
-   if (fClassInfo) {
```

Comment:
Hi Axel,

Well you may not have read the doc fully :).   Orginally 'fCanLoadClassInfo' was really indicating whether there was a chance to be able to load the ClassInfo or not  ... and it still does in a way.

At the moment when fCanLoadClassInfo is false, fClassInfo can be **either** a valid pointer **or** a **nullptr**, so "have loaded" would be inaccurate.

In other word,  if fCanLoadClassInfo is true running the body of LoadClassInfo is likely to yield a change in the value of fClassInfo, when it is false (and whether or not we already ran it) there is no point in running the body of LoadClassInfo as it is guaranteed to not yield any change in the value of fClassInfo.

Cheers,
Philippe.